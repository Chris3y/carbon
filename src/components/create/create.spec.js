import React from 'react';
import 'jest-styled-components';
import { shallow } from 'enzyme';
import TestRenderer from 'react-test-renderer';
import Create from './create.component';
import classicTheme from '../../style/themes/classic';
import CreateClassicStyle from './create-classic.style';
import CreateStyle from './create.style';

function render(props) {
  return shallow(<Create { ...props }> Create component </Create>);
}

describe('Create', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = render({});
  });

  it('should render correctly', () => {
    expect(TestRenderer.create(<CreateStyle />)).toMatchSnapshot();
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('when classic style has been provided', () => {
    it('should apply custom styling ', () => {
      wrapper.setProps({ theme: 'classic' });

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when `custom class` is provided to component', () => {
    it('should contain correct class name', () => {
      wrapper.setProps({ className: 'custom-class' });

      expect(wrapper.hasClass('custom-class')).toBeTruthy();
    });
  });
});

describe('Create classic', () => {
  expect(TestRenderer.create(<CreateClassicStyle theme={ classicTheme } />)).toMatchSnapshot();
});
