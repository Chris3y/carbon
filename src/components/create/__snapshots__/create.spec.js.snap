// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
"
    border: 1px dashed #99adb6;
    background-color: transparent;
    
    .carbon-icon::before {
        font-size:16px;
    }

    :hover{
        background: #fff;
    }

    :focus {
        color: #255BC7;
        border: 1px dashed #99adb6;
        outline: -webkit-focus-ring-color auto 5px;
    }
"
`;

exports[`Create should render correct 1`] = `
<Styled(Link)
  data-component="create"
  icon="add"
  iconAlign="right"
  theme={
    Object {
      "colors": Object {
        "border": "#668491",
        "brand": "#00DC00",
        "error": "#C7384F",
        "focus": "#FFB500",
        "focusedIcon": "#335B6D",
        "focusedLink": "#FFDA7F",
        "info": "#0073C2",
        "primary": "#26A826",
        "secondary": "#006300",
        "success": "#00B000",
        "tertiary": "#004400",
        "warning": "#E96400",
        "white": "#FFFFFF",
      },
      "disabled": Object {
        "background": "#E5EAEC",
        "border": "#CCD6DA",
        "disabled": "rgba(0,0,0,0.55)",
        "input": "#F2F4F5",
        "text": "rgba(0,0,0,0.3)",
      },
      "icon": Object {
        "focus": "#335B6D",
      },
      "name": "base",
      "shadows": Object {
        "cards": "0 3px 3px 0 rgba(0,20,29,0.2), 0 2px 4px 0 rgba(0,20,29,0.15)",
        "depth1": "0 5px 5px 0 rgba(0,20,29,0.2), 0 10px 10px 0 rgba(0,20,29,0.1)",
        "depth2": "0 10px 20px 0 rgba(0,20,29,0.2), 0 20px 40px 0 rgba(0,20,29,0.1)",
        "depth3": "0 10px 30px 0 rgba(0,20,29,0.1), 0 30px 60px 0 rgba(0,20,29,0.1)",
        "depth4": "0 10px 40px 0 rgba(0,20,29,0.04), 0 50px 80px 0 rgba(0,20,29,0.1)",
      },
      "text": Object {
        "color": "rgba(0,0,0,0.9)",
        "placeholder": "rgba(0,0,0,0.3)",
        "size": "14px",
      },
    }
  }
>
   Create component 
</Styled(Link)>
`;

exports[`Create should render correctly 1`] = `
.c2 {
  font-size: 14px;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #26A826;
}

.c2:hover {
  cursor: pointer;
  color: #006300;
}

.c2:focus {
  color: rgba(0,0,0,0.9);
  cursor: default;
  background-color: #FFDA7F;
  outline: none;
}

.c0 {
  display: inline-block;
}

.c1 {
  background-color: #F2F4F5;
  border: 1px dashed #668491;
  display: block;
  padding: 12px 12px 10px;
  text-align: center;
  font-weight: 700;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
}

.c1:hover {
  background-color: #FFFFFF;
}

.c1:focus {
  color: #26A826;
  background-color: #FFFFFF;
  outline: 3px solid #FFB500;
}

.c1 .carbon-icon::before {
  font-size: 12px;
}

<div
  className="c0"
  data-component="link"
  disabled={false}
>
  <a
    className="c1 c2"
    data-component="link"
    onKeyDown={[Function]}
    tabIndex="0"
  >
    <span>
      <span
        className="carbon-link__content"
      />
    </span>
  </a>
</div>
`;

exports[`Create when classic style has been provided should apply custom styling  1`] = `
<Styled(Link)
  data-component="create"
  icon="add"
  iconAlign="right"
  theme="classic"
>
   Create component 
</Styled(Link)>
`;
